extensions/VisualEditor/lib/ve/lib/jquery.client/jquery.client.js

```diff
@@ -0,0 +1,299 @@
+				platform = translate( match[1], platformTranslations );
+			}
+			if ( match = new RegExp( '(' + versionPrefixes.join( '|' ) + ')' + versionSuffix ).exec( ua ) ) {
+				version = match[3];
+			}
+
+			/* Edge Cases -- did I mention about how user agent string lie? */
+
+			// Decode Safari's crazy 400+ version numbers
+			if ( name === 'safari' && version > 400 ) {
+				version = '2.0';
+			}
+			// Expose Opera 10's lies about being Opera 9.8
+			if ( name === 'opera' && version >= 9.8 ) {
+				match = ua.match( /\bversion\/([0-9\.]*)/ );
+				if ( match && match[1] ) {
+					version = match[1];
+				} else {
+					version = '10';
+				}
+			}
+			// And Opera 15's lies about being Chrome
+			if ( name === 'chrome' && ( match = ua.match( /\bopr\/([0-9\.]*)/ ) ) ) {
+				if ( match[1] ) {
+					name = 'opera';
+					version = match[1];
+				}
+			}
+			// And IE 11's lies about being not being IE
+			if ( layout === 'trident' && layoutversion >= 7 && ( match = ua.match( /\brv[ :\/]([0-9\.]*)/ ) ) ) {
+				if ( match[1] ) {
+					name = 'msie';
+					version = match[1];
+				}
+			}
+			// And Amazon Silk's lies about being Android on mobile or Safari on desktop
+			if ( match = ua.match( /\bsilk\/([0-9.\-_]*)/ ) ) {
+				if ( match[1] ) {
+					name = 'silk';
+					version = match[1];
+				}
+			}
+
+			versionNumber = parseFloat( version, 10 ) || 0.0;
+
+			/* Caching */
+
+			return profileCache[ key  ] = {
+				name: name,
+				layout: layout,
+				layoutVersion: layoutversion,
+				platform: platform,
+				version: version,
+				versionBase: ( version !== x ? Math.floor( versionNumber ).toString() : x ),
+				versionNumber: versionNumber
+			};
+		},
+
+		/**
+		 * Checks the current browser against a support map object.
+		 *
+		 * Version numbers passed as numeric values will be compared like numbers (1.2 > 1.11).
+		 * Version numbers passed as string values will be compared using a simple component-wise
+		 * algorithm, similar to PHP's version_compare ('1.2' < '1.11').
+		 *
+		 * A browser map is in the following format:
+		 * {
+		 *   // Multiple rules with configurable operators
+		 *   'msie': [['>=', 7], ['!=', 9]],
+		 *    // Match no versions
+		 *   'iphone': false,
+		 *    // Match any version
+		 *   'android': null
+		 * }
+		 *
+		 * It can optionally be split into ltr/rtl sections:
+		 * {
+		 *   'ltr': {
+		 *     'android': null,
+		 *     'iphone': false
+		 *   },
+		 *   'rtl': {
+		 *     'android': false,
+		 *     // rules are not inherited from ltr
+		 *     'iphone': false
+		 *   }
+		 * }
+		 *
+		 * @param {Object} map Browser support map
+		 * @param {Object} [profile] A client-profile object
+		 * @param {boolean} [exactMatchOnly=false] Only return true if the browser is matched, otherwise
+		 * returns true if the browser is not found.
+		 *
+		 * @returns {boolean} The current browser is in the support map
+		 */
+		test: function ( map, profile, exactMatchOnly ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Function has a complexity of 24 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AFunctionComplexity)